components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOFacade.java

```diff
@@ -0,0 +1,331 @@
+import org.wso2.carbon.identity.webhook.management.internal.dao.WebhookManagementDAO;
+import org.wso2.carbon.identity.webhook.management.internal.util.WebhookManagementExceptionHandler;
+
+import java.util.List;
+
+/**
+ * Facade implementation for WebhookManagementDAO.
+ * This class coordinates between webhook subscriber services and webhook management DAO.
+ * It ensures that webhooks are first subscribed with external systems before being persisted.
+ */
+public class WebhookManagementDAOFacade implements WebhookManagementDAO {
+
+    private static final Log LOG = LogFactory.getLog(WebhookManagementDAOFacade.class);
+
+    private final WebhookManagementDAO webhookManagementDAO;
+
+    /**
+     * Constructor with WebhookManagementDAO.
+     *
+     * @param webhookManagementDAO WebhookManagementDAO instance.
+     */
+    public WebhookManagementDAOFacade(WebhookManagementDAO webhookManagementDAO) {
+
+        this.webhookManagementDAO = webhookManagementDAO;
+    }
+
+    /**
+     * Create a new webhook subscription in the database.
+     * First checks if a webhook with same endpoint exists, then subscribes with the webhook subscriber service,
+     * and finally persists it using the DAO.
+     *
+     * @param webhook  Webhook subscription to be created.
+     * @param tenantId Tenant ID.
+     * @throws WebhookMgtException If an error occurs while creating the webhook subscription.
+     */
+    @Override
+    public void createWebhook(Webhook webhook, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        EventSubscriberService subscriberService =
+                WebhookManagementComponentServiceHolder.getInstance().getEventSubscriberService();
+
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                boolean subscribed = subscriberService.subscribe(webhook, tenantDomain);
+                if (!subscribed) {
+                    LOG.warn("Failed to subscribe webhook: " + webhook.getId());
+                    webhook.setStatus("INACTIVE");
+                }
+                webhookManagementDAO.createWebhook(webhook, tenantId);
+                return null;
+            });
+        } catch (TransactionException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Error creating webhook: " + webhook.getId() +
+                        " in tenant ID: " + tenantId, e);
+            }
+            // If an error occurs after subscription but before adding to DB, attempt to unsubscribe
+            try {
+                subscriberService.unsubscribe(webhook, tenantDomain);
+            } catch (Exception ex) {
+                LOG.warn("Error unsubscribing webhook during rollback: ", ex);
+            }
```

Comment:
I think the implementation needs to be think though along with topic management. Let's keep it for the moment

Reply:
Address via https://github.com/wso2/product-is/issues/24084